app/routes/v2/add-standards-parcels/index.js

```diff
@@ -32,32 +21,66 @@ module.exports = [
   {
     method: 'POST',
     path: '/v2/add-standard-parcels',
-    handler: async (request, h) => {
-      const applyJourney = await cache.get('apply-journey', request.yar.id)
-      const parcels = applyJourney.selectedStandard.parcels
+    options: {
+      validate: {
+        payload: joi.object({
+          parcels: joi.any().required()
+        }),
+        failAction: async (request, h, error) => {
+          const { applyJourney, checkboxItems, totalHa } = await getAllItems(request)
+          return h.view('v2/add-standard-parcels/add-standard-parcels',
+            {
+              error: error,
+              checkboxItems: checkboxItems,
+              totalHa: Number(totalHa).toFixed(2),
+              sidebarItems: [`SBI: ${applyJourney.selectedSbi.sbi}`, `Standard: ${applyJourney.selectedStandard.name}`]
+            }).code(400).takeover()
+        }
+      },
+      handler: async (request, h) => {
+        const applyJourney = await cache.get('apply-journey', request.yar.id)
+        const parcels = applyJourney.selectedStandard.parcels
 
-      const selectedParcels = parcels.filter(item => request.payload.parcels.includes(item.id)).map(x => ({
-        id: x.id,
-        area: x.area
-      }))
+        const selectedParcels = parcels.filter(item => request.payload.parcels.includes(item.id)).map(x => ({
+          id: x.id,
+          area: x.area
+        }))
 
-      const parcelArea = selectedParcels.reduce((accum, item) => accum + item.area, 0)
+        const parcelArea = selectedParcels.reduce((accum, item) => accum + item.area, 0)
 
-      await cache.update('apply-journey', request.yar.id,
-        {
-          selectedParcels: selectedParcels,
-          parcelArea: Number(parcelArea).toFixed(2)
-        })
+        await cache.update('apply-journey', request.yar.id,
+          {
+            selectedParcels: selectedParcels,
+            parcelArea: Number(parcelArea).toFixed(2)
+          })
 
-      await sendAgreementCalculateMessage(
-        {
-          agreementNumber: applyJourney.agreementNumber,
-          callerId: applyJourney.callerId,
-          code: applyJourney.selectedStandard.code,
-          parcels: selectedParcels
-        }, request.yar.id)
+        await sendAgreementCalculateMessage(
+          {
+            agreementNumber: applyJourney.agreementNumber,
+            callerId: applyJourney.callerId,
+            code: applyJourney.selectedStandard.code,
+            parcels: selectedParcels
+          }, request.yar.id)
 
-      return h.redirect('/v2/choose-level')
+        return h.redirect('/v2/choose-level')
+      }
     }
   }
 ]
+
+const getAllItems = async (request) => {
+  const applyJourney = await cache.get('apply-journey', request.yar.id)
+  const parcels = applyJourney.selectedStandard.parcels
+
+  const checkboxItems = parcels.map(x => {
```

Comment:
Can simplify without the need of a return:

`const checkboxItems = parcels.map(x => (
    {
      text: `Parcel ${x.id} : ${Number(x.area).toFixed(2)}ha of ${applyJourney.selectedStandard.name}`,
      value: `${x.id}`,
      warnings: []
    })
)`